src/main/java/com/github/maleksandrowicz93/cqrsdemo/student/StudentWriteRepository.java

```diff
@@ -0,0 +1,9 @@
+package com.github.maleksandrowicz93.cqrsdemo.student;
+
+import com.github.maleksandrowicz93.cqrsdemo.interfaces.WriteRepository;
+
+import java.util.UUID;
+
+interface StudentWriteRepository extends WriteRepository<Student, UUID> {
+    boolean existsByEmail(String email);
```

Comment:
W jaki sposób "exists" to "write"? :p

Reply:
Podzieliłem to w taki sposób, żeby write handlary korzystały tylko z jednego (write repo), a query handler tylko z query repo (też repo z prelekcji zawierało Repo do zapisu z potrzebnymi metodami do odczytu mimo istnienia osobnego repo do odczytu). Z jednej strony zgadzam się, że takie metody niezgadzających się z intencję, którą przedstawia nazwa klasy, ale z drugiej przy dwóch osobnych bazach danych może wystąpić eventual consistency. Dzięki mojemu podejściu zyskujemy więc najnowsze dane. Pytanie więc, czy wolimy aktualne dane, czy czystszy kod, ale w sumie nie wiem, czy dbając o czysty kod powinienem myśleć o tym, jak działa baza danych, tym bardziej, że możemy podpinać różne konfiguracje i ta baza może być co minutę inna :P. Dodatkowo ten downtime pewnie też będzie marginalnie mały i ciężko byłoby manualnie zauważyć jego wpływ. Więc ostatecznie chyba się zgodzę z tym komentarzem :)